### PR TITLE
Document custom protobuf msgs/queries

### LINKED ISSUE #640: Document how to generate protobuf codec file and use it
This allows us to add support in an app for key modules that are not already in @cosmjs/stargate, or any custom chain-specific modules (which will never be in @cosmjs/stargate). This seems quite automated, but no one outside the cosmjs team knows how to do it.

Documenting this is an alternative solution to #636

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #640

I wasn’t sure where to put this file. Should it be in the `@cosmjs/stargate` package?

### CODE REVIEW

File: docs/CUSTOM_PROTOBUF_CODECS.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  `@cosmjs/tendermint-rpc` as dependencies.
+- You have installed `ts-proto` as a development dependency.
+- You have installed [`protoc`](https://github.com/protocolbuffers/protobuf).
+- This document assumes that the protocol buffer definitions which you need are
+  already available somewhere in `.proto` files.
+
+## Step 1: Acquire the definition files
+
+You will need these files locally. For example, we use
+[this script](https://github.com/cosmos/cosmjs/blob/d8cdd83/packages/stargate/scripts/get-proto.sh)
```
Comment (webmaster128) [inline docs/CUSTOM_PROTOBUF_CODECS.md:27]:
```suggestion
[this script](https://github.com/cosmos/cosmjs/blob/v0.25.0-alpha.0/packages/stargate/scripts/get-proto.sh)
```

File: docs/CUSTOM_PROTOBUF_CODECS.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+is an example usage:
+
+```sh
+protoc \
+  --plugin="./node_modules/.bin/protoc-gen-ts_proto" \
+  --ts_proto_out="./path/to/output/directory" \
+  --proto_path="./path/to/definitions" \
+  --ts_proto_opt="esModuleInterop=true,forceLong=long,useOptionals=true" \
+  "./path/to/definitions/file.proto" \
+  "./path/to/definitions/another.proto
```
Comment (webmaster128) [inline docs/CUSTOM_PROTOBUF_CODECS.md:43]:
```suggestion
  "./path/to/definitions/another.proto"
```

File: docs/CUSTOM_PROTOBUF_CODECS.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+```
+
+Note that the available `ts-proto` options are described
+[here](https://github.com/stephenh/ts-proto#supported-options). You can see the
+script we use for the `@cosmjs/stargate` package
+[here](https://github.com/cosmos/cosmjs/blob/d8cdd83/packages/stargate/scripts/define-proto.sh).
```
Comment (webmaster128) [inline docs/CUSTOM_PROTOBUF_CODECS.md:49]:
```suggestion
[here](https://github.com/cosmos/cosmjs/blob/v0.25.0-alpha.0/packages/stargate/scripts/define-proto.sh).
```

File: docs/CUSTOM_PROTOBUF_CODECS.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+const signer = await DirectSecp256k1HdWallet.fromMnemonic(
+  mnemonic,
+  undefined,
+  "myprefix",
+);
+const options = { registry: myRegistry };
+const client = await SigningStargateClient.connectWithSigner(
+  "my.endpoint.com",
+  signer,
+  options, // Here you provide the custom registry
```
Comment (webmaster128) [inline docs/CUSTOM_PROTOBUF_CODECS.md:82]: Maybe easier to follow

```suggestion
const client = await SigningStargateClient.connectWithSigner(
  "my.endpoint.com",
  signer,
  { registry: myRegistry },
```

**Review by webmaster128 — COMMENTED** [review]
Very nice. I'd move this to the stargate package

**Review by webmaster128 — APPROVED** [review]

**Review by ethanfrey — COMMENTED** [review]
Very nice.

Only addition from me would be to link this prominently from the README. There should be a "using stargate" section or similar

**Comment by willclarktech** [comment]
@ethanfrey Covered in https://github.com/cosmos/cosmjs/pull/715/files

**Pull request merged**